core/src/main/java/org/elasticsearch/index/query/FuzzyQueryParser.java

```diff
@@ -60,23 +68,23 @@ public FuzzyQueryBuilder fromXContent(QueryParseContext parseContext) throws IOE
                 if (token == XContentParser.Token.FIELD_NAME) {
                     currentFieldName = parser.currentName();
                 } else {
-                    if ("term".equals(currentFieldName)) {
+                    if (parseContext.parseFieldMatcher().match(currentFieldName, TERM_FIELD)) {
```

Comment:
Before actually making the change for all classes, what do you think of the following:

    diff --git a/core/src/main/java/org/elasticsearch/index/query/FuzzyQueryParser.java b/core/src/main/java/org/elasticsearch/index/query/FuzzyQueryParser.java
    index 85365f8..e0120c1 100644
    --- a/core/src/main/java/org/elasticsearch/index/query/FuzzyQueryParser.java
    +++ b/core/src/main/java/org/elasticsearch/index/query/FuzzyQueryParser.java
    @@ -68,23 +68,23 @@ public class FuzzyQueryParser implements QueryParser<FuzzyQueryBuilder> {
                     if (token == XContentParser.Token.FIELD_NAME) {
                         currentFieldName = parser.currentName();
                     } else {
    -                    if (parseContext.parseFieldMatcher().match(currentFieldName, TERM_FIELD)) {
    +                    if (match(currentFieldName, TERM_FIELD, parseContext)) {
                             value = parser.objectBytes();
    -                    } else if (parseContext.parseFieldMatcher().match(currentFieldName, VALUE_FIELD)) {
    +                    } else if (match(currentFieldName, VALUE_FIELD, parseContext)) {
                             value = parser.objectBytes();
    -                    } else if (parseContext.parseFieldMatcher().match(currentFieldName, AbstractQueryBuilder.BOOST_FIELD)) {
    +                    } else if (match(currentFieldName, AbstractQueryBuilder.BOOST_FIELD, parseContext)) {
                             boost = parser.floatValue();
    -                    } else if (parseContext.parseFieldMatcher().match(currentFieldName, Fuzziness.FIELD)) {
    +                    } else if (match(currentFieldName, Fuzziness.FIELD, parseContext)) {
                             fuzziness = Fuzziness.parse(parser);
    -                    } else if (parseContext.parseFieldMatcher().match(currentFieldName, PREFIX_LENGTH_FIELD)) {
    +                    } else if (match(currentFieldName, PREFIX_LENGTH_FIELD, parseContext)) {
                             prefixLength = parser.intValue();
    -                    } else if (parseContext.parseFieldMatcher().match(currentFieldName, MAX_EXPANSIONS_FIELD)) {
    +                    } else if (match(currentFieldName, MAX_EXPANSIONS_FIELD, parseContext)) {
                             maxExpansions = parser.intValue();
    -                    } else if (parseContext.parseFieldMatcher().match(currentFieldName, TRANSPOSITIONS_FIELD)) {
    +                    } else if (match(currentFieldName, TRANSPOSITIONS_FIELD, parseContext)) {
                             transpositions = parser.booleanValue();
    -                    } else if (parseContext.parseFieldMatcher().match(currentFieldName, REWRITE_FIELD)) {
    +                    } else if (match(currentFieldName, REWRITE_FIELD, parseContext)) {
                             rewrite = parser.textOrNull();
    -                    } else if (parseContext.parseFieldMatcher().match(currentFieldName, AbstractQueryBuilder.NAME_FIELD)) {
    +                    } else if (match(currentFieldName, AbstractQueryBuilder.NAME_FIELD, parseContext)) {
                             queryName = parser.text();
                         } else {
                             throw new ParsingException(parser.getTokenLocation(), "[fuzzy] query does not support [" + currentFieldName + "]");
    diff --git a/core/src/main/java/org/elasticsearch/index/query/QueryParser.java b/core/src/main/java/org/elasticsearch/index/query/QueryParser.java
    index 0a3f6d6..4737c4e 100644
    --- a/core/src/main/java/org/elasticsearch/index/query/QueryParser.java
    +++ b/core/src/main/java/org/elasticsearch/index/query/QueryParser.java
    @@ -19,6 +19,8 @@
     
     package org.elasticsearch.index.query;
     
    +import org.elasticsearch.common.ParseField;
    +
     import java.io.IOException;
     
     /**
    @@ -48,4 +50,11 @@ public interface QueryParser<QB extends QueryBuilder<QB>> {
          * @return an empty {@link QueryBuilder} instance for this parser that can be used for deserialization
          */
         QB getBuilderPrototype();
    +    
    +    /**
    +     * @return true if the mentioned currentFieldName matches matchField
    +     * */
    +    default boolean match(String currentFieldName, ParseField matchField, QueryParseContext parseContext) {
    +       return parseContext.parseFieldMatcher().match(currentFieldName, matchField);
    +    }
     }